src/routes/__tests__/routes.test.tsx

```diff
@@ -60,18 +75,52 @@ describe('routes', () => {
   })
 
   describe('/pensjon/kalkulator/samtykke', () => {
-    it('viser Steg 2', () => {
+    it('redirigerer til Step 1 når brukeren prøver å aksessere steget med direkte url', async () => {
       const router = createMemoryRouter(routes, {
         basename: ROUTER_BASE_URL,
         initialEntries: ['/pensjon/kalkulator/samtykke'],
       })
-      render(<RouterProvider router={router} />, { hasRouter: false })
-      expect(screen.getByText('stegvisning.samtykke.title')).toBeInTheDocument()
+      render(<RouterProvider router={router} />, {
+        hasRouter: false,
+      })
+      expect(
+        await screen.findByText('stegvisning.start.start')
+      ).toBeInTheDocument()
+    })
+
+    it('viser Steg 2 (gitt at brukeren kommer til steget gjennom stegvisningen)', async () => {
+      const mockedState = {
+        api: {
+          ...fakeApiCalls,
+        },
+        userInput: { ...userInputInitialState },
+      }
+      store.getState = vi.fn().mockImplementation(() => {
+        return mockedState
+      })
+      const router = createMemoryRouter(routes, {
+        basename: ROUTER_BASE_URL,
+        initialEntries: ['/pensjon/kalkulator/samtykke'],
+      })
+      render(<RouterProvider router={router} />, {
+        preloadedState: mockedState as unknown as RootState,
```

Comment:
Ser ikke ut som du trenger preloadedState her siden du uansett mocker kallet til getState

Reply:
Takk, `preloadedState` henger fra før. Har ryddet opp nå.
Jeg har faktisk først prøvd tilnærmingen med store som argument i loader for så å ende opp med denne løsningen her. Det var mer jobb å tilpasse testene, men selve koden i `routes` og `loader'en` ble mye ryddigere av det.